lib/matplotlib/axes/_base.py

```diff
@@ -3969,10 +3969,26 @@ def format_ydata(self, y):
 
     def format_coord(self, x, y):
         """Return a format string formatting the *x*, *y* coordinates."""
-        return "x={} y={}".format(
-            "???" if x is None else self.format_xdata(x),
-            "???" if y is None else self.format_ydata(y),
-        )
+        twins = self._twinned_axes.get_siblings(self)
+        if len(twins) == 1:
+            return "(x, y) = ({}, {})".format(
+                "???" if x is None else self.format_xdata(x),
+                "???" if y is None else self.format_ydata(y))
+        screen_xy = self.transData.transform((x, y))
+        xy_strs = []
+        # Retrieve twins in the order of self.figure.axes to sort tied zorders (which is
+        # the common case) by the order in which they are added to the figure.
+        for ax in sorted(
+            [ax for ax in self.figure.axes if (
+                ax in twins and (
+                    ax is self
+                    or ax._sharex is self or self._sharex is ax
+                    or ax._sharey is self or self._sharey is ax))],
```

Comment:
What is this second part of the condition for?

Reply:
When two axes are shared, only one has self._sharex set to the other axes (but not the other way round), so we have to check both attributes to find the pairs.

Reply:
No, I understood that; I meant the whole highlighted section after the first `and`.

Reply:
Do we have an example of an ax being in the twins list that is not actually a twin?

Reply:
Ping? Branching tomorrow, if I can.

Reply:
That turned out to be just leftover code from before I made Grouper handling the ordering; I agree the whole thing is unneeded now, removed it.